src/blocks/store/s3.clj

```diff
@@ -221,9 +234,12 @@
         (object->block client bucket prefix stats)
         ; Otherwise, upload block to S3.
         (let [object-key (id->key prefix (:id block))
-              ; TODO: support function which allows you to mutate the ObjectMetadata before it is sent in the putObject call?
               metadata (doto (ObjectMetadata.)
                          (.setContentLength (:size block)))]
+          (when sse-algorithm
+            (.setSSEAlgorithm metadata sse-algorithm))
```

Comment:
This assumes the record attribute is a string constant here, but you're setting `:sse` in the constructor instead of `sse-algorithm`. I think it'd be more friendly to leave the attribute as a keyword and do the resolution in this call. For example, nothing prevents the user from associng an invalid value into the record, which a construction-time check wouldn't catch.

Reply:
Whoops missed that in my renaming of `:sse-algorithm` to `:sse`. I'll change the keyword value check to the constructor then the lookup here.